hector_mapping/src/HectorMappingRos.cpp

```diff
@@ -231,106 +231,100 @@ HectorMappingRos::~HectorMappingRos()
     delete map__publish_thread_;
 }
 
-void HectorMappingRos::scanCallback(const sensor_msgs::LaserScan& scan)
-{
+void HectorMappingRos::scanCallback(const sensor_msgs::LaserScan& scan) {
   if (pause_scan_processing_) {
     return;
   }
 
-  if (hectorDrawings)
-  {
+  if (hectorDrawings) {
     hectorDrawings->setTime(scan.header.stamp);
   }
 
-  ros::WallTime startTime = ros::WallTime::now();
-
-  if (!p_use_tf_scan_transformation_)
-  {
-    if (rosLaserScanToDataContainer(scan, laserScanContainer,slamProcessor->getScaleToMap()))
-    {
+  ros::WallTime start_time = ros::WallTime::now();
+  if (!p_use_tf_scan_transformation_) {
+    // If we are not using the tf tree to find the transform between the base frame and laser frame,
+    // then just convert the laser scan to our data container and process the update based on our last
+    // pose estimate
+    if (rosLaserScanToDataContainer(scan, laserScanContainer,slamProcessor->getScaleToMap())) {
       slamProcessor->update(laserScanContainer,slamProcessor->getLastScanMatchPose());
     }
-  }
-  else
-  {
-    ros::Duration dur (0.5);
-
-    if (tf_.waitForTransform(p_base_frame_,scan.header.frame_id, scan.header.stamp,dur))
-    {
-      tf::StampedTransform laserTransform;
+  } else {
+    // If we are using the tf tree to find the transform between the base frame and laser frame,
+    // let's get that transform
+    const ros::Duration dur (0.5);
+    tf::StampedTransform laserTransform;
+    if (tf_.waitForTransform(p_base_frame_,scan.header.frame_id, scan.header.stamp,dur)) {
       tf_.lookupTransform(p_base_frame_,scan.header.frame_id, scan.header.stamp, laserTransform);
+    } else {
+      ROS_INFO("lookupTransform %s to %s timed out. Could not transform laser scan into base_frame.", p_base_frame_.c_str(), scan.header.frame_id.c_str());
+      return;
+    }
 
-      //projector_.transformLaserScanToPointCloud(p_base_frame_ ,scan, pointCloud,tf_);
-      projector_.projectLaser(scan, laser_point_cloud_,30.0);
+    // Convert the laser scan to point cloud
+    projector_.projectLaser(scan, laser_point_cloud_, 30.0);
 
-      if (scan_point_cloud_publisher_.getNumSubscribers() > 0){
-        scan_point_cloud_publisher_.publish(laser_point_cloud_);
-      }
+    // Publish the point cloud if there are any subscribers
+    if (scan_point_cloud_publisher_.getNumSubscribers() > 0) {
+      scan_point_cloud_publisher_.publish(laser_point_cloud_);
+    }
 
-      Eigen::Vector3f startEstimate(Eigen::Vector3f::Zero());
+    // Return if we can't convert the point cloud to our data container
+    if(!rosPointCloudToDataContainer(laser_point_cloud_, laserTransform, laserScanContainer, slamProcessor->getScaleToMap())) {
+      return;
+    }
```

Comment:
honestly, I don't see the point in checking the return value of `rosPointCloudToDataContainer`, since it always returns true... maybe it would be better to make `rosPointCloudToDataContainer` a `void` function, instead of `bool`, and don't even check the return value